crates/transaction-emitter-lib/src/transaction_generator/account_generator.rs

```diff
@@ -44,31 +63,62 @@ impl TransactionGenerator for AccountGenerator {
         &mut self,
         accounts: Vec<&mut LocalAccount>,
         transactions_per_account: usize,
-        _all_addresses: Arc<Vec<AccountAddress>>,
-        _invalid_transaction_ratio: usize,
-        gas_price: u64,
     ) -> Vec<SignedTransaction> {
         let mut requests = Vec::with_capacity(accounts.len() * transactions_per_account);
+        let mut new_accounts = Vec::with_capacity(accounts.len() * transactions_per_account);
         for account in accounts {
             for _ in 0..transactions_per_account {
                 let receiver = LocalAccount::generate(&mut self.rng).address();
                 let request =
-                    self.gen_single_txn(account, receiver, 0, &self.txn_factory, gas_price);
+                    self.gen_single_txn(account, receiver, &self.txn_factory, self.gas_price);
                 requests.push(request);
+                new_accounts.push(receiver);
+            }
+        }
+
+        if self.add_created_accounts_to_pool {
+            let mut current = self.all_addresses.write();
+            if current.len() < self.max_working_set {
+                current.append(&mut new_accounts);
+                sample!(
+                    SampleRate::Duration(Duration::from_secs(15)),
+                    info!("Accounts working set increased to {}", current.len())
+                );
+            } else {
+                info!(
+                    "Already at limit {} > {}",
```

Comment:
Hmm..I am wondering if a better behavior would be to replace some existing account in the list? That way, we can ensure the working set doesn't become static after hitting this limit.

Reply:
yeah, I was thinking about that :)